ga4gh/datarepo.py

```diff
@@ -179,42 +262,826 @@ def __init__(
             self.addDataset(dataset)
 
 
-class FileSystemDataRepository(AbstractDataRepository):
+class SqlDataRepository(AbstractDataRepository):
     """
-    A data repository based on the file system
+    A data repository based on a SQL database.
     """
-    referenceSetsDirName = "referenceSets"
-    datasetsDirName = "datasets"
-    ontologiesDirName = "ontologymaps"
-
-    def __init__(self, dataDir):
-        super(FileSystemDataRepository, self).__init__()
-        self._dataDir = dataDir
-        sourceDirNames = [self.referenceSetsDirName,
-                          self.ontologiesDirName,
-                          self.datasetsDirName]
-        constructors = [
-            references.HtslibReferenceSet, ontologies.FileSystemOntologies,
-            datasets.FileSystemDataset]
-        objectAdders = [self.addReferenceSet, self.addOntologyMap,
-                        self.addDataset]
-        for sourceDirName, constructor, objectAdder in zip(
-                sourceDirNames, constructors, objectAdders):
-            sourceDir = os.path.join(self._dataDir, sourceDirName)
-            for setName in os.listdir(sourceDir):
-                relativePath = os.path.join(sourceDir, setName)
-                if os.path.isdir(relativePath):
-                    objectAdder(constructor(setName, relativePath, self))
-
-    def checkConsistency(self):
-        """
-        Perform checks that ensure the consistency of the data.
-        Factored into a separate method from server init since the
-        data repo object can be created on a partially-complete
-        data set.
+    class SchemaVersion(object):
+        """
+        The version of the data repository SQL schema
+        """
+        def __init__(self, versionString):
+            splits = versionString.split('.')
+            assert len(splits) == 2
+            self.major = splits[0]
+            self.minor = splits[1]
+
+        def __str__(self):
+            return "{}.{}".format(self.major, self.minor)
+
+    version = SchemaVersion("0.1")
+    systemKeySchemaVersion = "schemaVersion"
+    systemKeyCreationTimeStamp = "creationTimeStamp"
+
+    def __init__(self, fileName):
+        super(SqlDataRepository, self).__init__()
+        self._dbFilename = fileName
```

Comment:
Does it make sense to enforce, or add a `.db` extension?

`repo init repo`, `repo init repo.db`